doc: Improve Pod Standards doc (#5905)
operator-framework/operator-sdk: website/content/en/docs/best-practices/pod-security-standards.md

```

#### How do I check what policy (Privileged, Baseline and Restricted) is configured for my Operator and Operand(s)?

An easy way might be using the tool: [psachecker][psachecker]. This tool is only
able to be used to check locally the `Deployments/Pods` manifests and not the CSV.

Alternatively, you could test the policy enforcement by labeling the namespaces and running your operator or applying the manifests. [More info][enforce-standards-namespace-labels].

##### How do I install psachecker?

The following steps will install [psachecker][psachecker] in a Golang environment.

```sh
git clone git@github.com:stlaz/psachecker.git $GOPATH/src/github.com/stlaz/psachecker
cd $GOPATH/src/github.com/stlaz/psachecker
make build
cp kubectl-psachecker $GOPATH/bin/
```

> #### How to use psachecker to verify the policy of a manifests?

Reviewer: Nit: 
```suggestion
#### How do I use psachecker to verify the policy of a manifests?
```

    pod-security.kubernetes.io/warn: privileged
```

Following you will find a detailed description of how to configure and test your solutions. 
The most straightforward way to ensure if your workloads can work on a restricted namespace is verifing if your solution can run in namespaces enforced as restricted. 

> **RECOMMENDED** ensure the desired behaviour via the e2e test. (_PS.:You will find code examples on the content shared_)

Reviewer: Nit:
```suggestion
**NOTE**: It is recommended that you test the desired behavior as part of an e2e test suite. Examples of an e2e test for this will be shown in a later section.
```

### How can I verify my manifest? 

#### Using Kind

To verify the policy of your Pod/Container(s) you might to use a [Kind](https://kind.sigs.k8s.io/docs/user/quick-start/)
cluster as described in the [K8s documentation](https://kubernetes.io/docs/tutorials/security/cluster-level-pss/).

**Example**

1. First lets create a namespace where we will test the files:

```sh
kubectl create ns mytest
```

2. Now, let's label the namespace so that we can check our manifest

```sh
 kubectl label --overwrite ns mytest \
   pod-security.kubernetes.io/enforce=restricted \
   pod-security.kubernetes.io/enforce-version=v1.24 \
   pod-security.kubernetes.io/audit: restricted
```

3. Now, apply the following Pod which is **not** restricted on the namespace:

```yaml
apiVersion: v1
kind: Pod
metadata:
 name: example
 namespace: mytest
spec:
 containers:
  - name: test
    securityContext:
     # see that we are allowing privilege escalation  
     allowPrivilegeEscalation: true 
    image: 'busybox:1.28'
    ports:
     - containerPort: 8080

```

> 4. Then, when we try to apply the Pod manifest above we will check:

Reviewer: Nit:
```suggestion
4. Then, when we try to apply the Pod manifest above we should see an error:
```

```sh
$ kubectl apply -f mypodtest.yaml 
Error from server (Forbidden): error when creating "mypodtest.yaml": pods "example" is forbidden: violates PodSecurity "restricted:v1.24": allowPrivilegeEscalation != false (container "test" must set securityContext.allowPrivilegeEscalation=false), unrestricted capabilities (container "test" must set securityContext.capabilities.drop=["ALL"]), runAsNonRoot != true (pod or container "test" must set securityContext.runAsNonRoot=true), seccompProfile (pod or container "test" must set securityContext.seccompProfile.type to "RuntimeDefault" or "Localhost")
```

> #### How do I check what is the policy(Privileged, Baseline and Restricted) configured for my Operator and Operand(s)?

Reviewer: Nit:
```suggestion
#### How do I check what policy (Privileged, Baseline and Restricted) is configured for my Operator and Operand(s)?
```

An easy way might be using the tool: [psachecker][psachecker]. This tool is only
able to be used to check locally the `Deployments/Pods` manifests and not the CSV.

> However, you could also do the test be enforce the policies lebeling the namespaces 
> and running your solutions there or applying the manifests. [More info][enforce-standards-namespace-labels].

Reviewer: Nit:
```suggestion
Alternatively, you could test the policy enforcement by labeling the namespaces and running your operator or applying the manifests. [More info][enforce-standards-namespace-labels].
```

> ##### How to install psachecker?

Reviewer: Nit:
```suggestion
##### How do I install psachecker?
```

> Following the steps to install the [psachecker][psachecker] in an env with Golang.

Reviewer: Nit:
```suggestion
The following steps will install [psachecker][psachecker] in a Golang environment.
```

```sh
git clone git@github.com:stlaz/psachecker.git $GOPATH/src/github.com/stlaz/psachecker
cd $GOPATH/src/github.com/stlaz/psachecker
make build
cp kubectl-psachecker $GOPATH/bin/
```

#### How to use psachecker to verify the policy of a manifests?

> In this example, let's see how we can check how can we verify our Operator and the deployments configured
> via the CSV. Following the steps.

Reviewer: Nit:
```suggestion
The following steps will demonstrate how we can use [psachecker][psachecker] to verify that our Operator and deployments are properly configured in the CSV.
```

which defines the Pod/Container(s) that will be installed by OLM to get your Operator running on the cluster. 
In order for the security standards to be followed you will need to ensure the configurations are set correctly.

**Note: Ensure the configuration is carried to the Pod/Containers on the bundle CSV after running make bundle**. See that 
the Operator bundle generated with the target is built from the manifests under the `config` directory. To know more about

> the layout of your operator built with Operator-SDK see the doc [Project Layout](../../docs/overview/project-layout.md).

Author: ```suggestion
the layout of your operator built with Operator-SDK see the doc [Project Layout](/docs/overview/project-layout.md).
```

To check an example of a CSV which complies with the [restricted][restricted] policy, see the Golang sample
under the [testdata/go/v3/memcached-operator/bundle/manifests/memcached-operator.clusterserviceversion.yaml](https://github.com/operator-framework/operator-sdk/blob/master/testdata/go/v3/memcached-operator/bundle/manifests/memcached-operator.clusterserviceversion.yaml)

### How can I verify my manifest? How do I check what policy is configured?

#### Using Kind

To verify the policy of your Pod/Container(s) you might to use a [Kind](https://kind.sigs.k8s.io/docs/user/quick-start/)
cluster as described in the [K8s documentation](https://kubernetes.io/docs/tutorials/security/cluster-level-pss/).

**Example**

1. First lets create a namespace where we will test the files:

```sh
kubectl create ns mytest
```

2. Now, let's label the namespace so that we can check our manifest

```sh

>  kubectl label --overwrite ns mytest pod-security.kubernetes.io/enforce=restricted   pod-security.kubernetes.io/enforce-version=v1.24

Author: ```suggestion
 kubectl label --overwrite ns mytest \
   pod-security.kubernetes.io/enforce=restricted \
   pod-security.kubernetes.io/enforce-version=v1.24 \
   pod-security.kubernetes.io/audit: restricted
```

which defines the Pod/Container(s) that will be installed by OLM to get your Operator running on the cluster. 
In order for the security standards to be followed you will need to ensure the configurations are set correctly.

**Note: Ensure the configuration is carried to the Pod/Containers on the bundle CSV after running make bundle**. See that 
the Operator bundle generated with the target is built from the manifests under the `config` directory. To know more about

> the layout of your operator built with Operator-SDK see the doc [Project Layout](/docs/overview/project-layout.md).

Author: ```suggestion
the layout of your operator built with Operator-SDK see [Project Layout][project-layout].
```

- Then, you can run the tool and check if its result will be `restricted` as expected (i.e.):

```sh
$ kubectl-psachecker inspect-workloads -f test.yaml
system: restricted

> ```

Author: ```suggestion
```

### Can I use the metrics to check if my Pod/Containers are violating the PodSecurity policies?

Yes, you can. You need to label the namespaces with `pod-security.kubernetes.io/audit: restricted` (i.e. `kubectl label --overwrite ns --all pod-security.kubernetes.io/enforce-version=v1.24 pod-security.kubernetes.io/audit: restricted`).  But note that you can check results that do not come from your Operator and Operands. If you are looking to use the metrics to do the checks then, ensure that you check its results before and after performing the tests, see an example: 

```sh
kubectl get --raw /metrics | prom2json | jq '[.[] | select(.name=="pod_security_evaluations_total") ]'
```
```json
[
  {
    "name": "pod_security_evaluations_total",
    "help": "[ALPHA] Number of policy evaluations that occurred, not counting ignored or exempt requests.",
    "type": "COUNTER",
    "metrics": [
      {
        "labels": {
          "decision": "allow",
          "mode": "enforce",
          "policy_level": "privileged",
          "policy_version": "latest",
          "request_operation": "create",
          "resource": "pod",
          "subresource": ""
        },
        "value": "29"
      },
      {
        "labels": {
          "decision": "allow",
          "mode": "enforce",
          "policy_level": "privileged",
          "policy_version": "latest",
          "request_operation": "update",
          "resource": "pod",
          "subresource": ""
        },
        "value": "0"
      },
      {
        "labels": {
          "decision": "deny",
          "mode": "audit",
          "policy_level": "restricted",
          "policy_version": "latest",
          "request_operation": "create",
          "resource": "controller",
          "subresource": ""
        },
        "value": "6"
      },
      {
        "labels": {
          "decision": "deny",
          "mode": "audit",
          "policy_level": "restricted",
          "policy_version": "latest",
          "request_operation": "create",
          "resource": "pod",
          "subresource": ""
        },
        "value": "7"
      },
      {
        "labels": {
          "decision": "deny",
          "mode": "warn",
          "policy_level": "restricted",
          "policy_version": "latest",
          "request_operation": "create",
          "resource": "controller",
          "subresource": ""
        },
        "value": "6"
      },
      {
        "labels": {
          "decision": "deny",
          "mode": "warn",
          "policy_level": "restricted",
          "policy_version": "latest",
          "request_operation": "create",
          "resource": "pod",
          "subresource": ""
        },
        "value": "7"
      }
    ]
  }
]
```
Also, you might be able to create Prometheus alerts such as:

```
sum (increase(pod_security_evaluations_total{decision="deny",mode="audit"}[1h])) by (policy_level)
```

```

- Then, you can run the tool and check if its result will be `restricted` as expected (i.e.):

```sh
$ kubectl-psachecker inspect-workloads -f test.yaml
system: restricted
```

> 

Author: ```suggestion
[project-layout]: /docs/overview/project-layout
```

- Then, you can run the tool and check if its result will be `restricted` as expected (i.e.):

```sh
$ kubectl-psachecker inspect-workloads -f test.yaml
system: restricted
```

### Can I use the metrics to check if my Pod/Containers are violating the PodSecurity policies?

Yes, you can. You need to label the namespaces with `pod-security.kubernetes.io/audit: restricted` 
(i.e. `kubectl label --overwrite ns --all pod-security.kubernetes.io/enforce-version=v1.24 pod-security.kubernetes.io/audit: restricted`).

> But note that you can check results that do not come from your Operator and Operand(s). 

Reviewer: Nit:
```suggestion
 It is important to note that the results may include metrics that do not come from your Operator and Operand(s). 
```

> If you are looking to use the metrics to do the checks then, ensure that you check its 
> results before and after performing the tests, see an example: 

Reviewer: Nit:
```suggestion
If you are looking to use the metrics to do the checks, ensure that you check the
results before and after performing the tests, for example: 
```

          "subresource": ""
        },
        "value": "6"
      },
      {
        "labels": {
          "decision": "deny",
          "mode": "audit",
          "policy_level": "restricted",
          "policy_version": "latest",
          "request_operation": "create",
          "resource": "pod",
          "subresource": ""
        },
        "value": "7"
      },
      {
        "labels": {
          "decision": "deny",
          "mode": "warn",
          "policy_level": "restricted",
          "policy_version": "latest",
          "request_operation": "create",
          "resource": "controller",
          "subresource": ""
        },
        "value": "6"
      },
      {
        "labels": {
          "decision": "deny",
          "mode": "warn",
          "policy_level": "restricted",
          "policy_version": "latest",
          "request_operation": "create",
          "resource": "pod",
          "subresource": ""
        },
        "value": "7"
      }
    ]
  }
]
```
Also, you might be able to create Prometheus alerts such as:

```sh
sum (increase(pod_security_evaluations_total{decision="deny",mode="audit"}[1h])) by (policy_level)
```

> ### (Recommended) Now, how could I automate this checks (e2e) to ensure that my solutions can run under the policies?

Reviewer: Nit:
```suggestion
### (Recommended) How can I automate this check using e2e testing to ensure that my solutions can run under the policies?
```

> In the tests, you can create and label the namespace with the desired policy (i.e. restricted). So that you can verify if you 
> will check warnings and/or if the Pod/Containers are `Running` successfully. It can be something like:

Reviewer: Nit:
```suggestion
In the tests, you can create and label a namespace with the desired policy (i.e. restricted). This would allow you to verify
if any warnings/errors occur and if the Pod(s)/Container(s) are in the `Running` state. It can be something like:
```
